modules/javafx.graphics/src/main/native-glass/gtk/glass_window.cpp

```diff
@@ -1010,83 +1004,54 @@ void WindowContextTop::process_net_wm_property() {
-#ifdef GLASS_GTK3
-        gdk_window_get_geometry(gdk_window, NULL, NULL, &w, &h);
-#else
-        gdk_window_get_geometry(gdk_window, NULL, NULL, &w, &h, NULL);
-#endif
-        x = frame.x;
-        y = frame.y;
-        geometry.current_width = frame.width;
-        geometry.current_height = frame.height;
-
-        if (update_frame_extents()) {
-            updateWindowConstraints = true;
-            if (!frame_extents_initialized && !is_null_extents()) {
-                frame_extents_initialized = true;
-                set_bounds(0, 0, false, false,
-                    requested_bounds.width, requested_bounds.height,
-                    requested_bounds.client_width, requested_bounds.client_height
-                );
-            }
-        }
-    } else {
-        x = event->x;
-        y = event->y;
-        w = event->width;
-        h = event->height;
+    // This will prevent sending window sizes before java is done configuring the window
+    // Without it, some content size oriented window gets the size before finishing the layout
+    if (!map_received && !is_fullscreen && !is_maximized) {
+        return;
     }
 
-    if (size_assigned && w <= 1 && h <= 1 && (geometry.final_width.value > 1 ||
-                                             geometry.final_height.value > 1)) {
-        // skip artifact
-        return;
-   }
+    int x, y, w, h;
+
+    x = event->x - geometry.extents.left;
+    y = event->y - geometry.extents.top;
+    w = event->width;
+    h = event->height;
 
-    // JDK-8232811: to avoid conflicting events, update the geometry only after window pops.
-    if (map_received) {
+    bool size_changed = geometry.final_width.value != w
+                        || geometry.final_height.value != h;
+
+    if (size_changed) {
         geometry.final_width.value = w;
         geometry.final_width.type = BOUNDSTYPE_CONTENT;
         geometry.final_height.value = h;
         geometry.final_height.type = BOUNDSTYPE_CONTENT;
+
+        if (jwindow) {
+            notify_window_resize();
+        }
     }
 
-    geometry_set_window_x(&geometry, x);
-    geometry_set_window_y(&geometry, y);
+    bool moved = (geometry.current_x != x || geometry.current_y != y);
 
-    if (jview) {
-        mainEnv->CallVoidMethod(jview, jViewNotifyResize,
-                event->width,
-                event->height);
-        CHECK_JNI_EXCEPTION(mainEnv)
-        mainEnv->CallVoidMethod(jview, jViewNotifyView,
-                com_sun_glass_events_ViewEvent_MOVE);
-        CHECK_JNI_EXCEPTION(mainEnv)
-    }
-    if (jwindow) {
-        mainEnv->CallVoidMethod(jwindow, jWindowNotifyResize,
-                (is_maximized)
-                    ? com_sun_glass_events_WindowEvent_MAXIMIZE
-                    : com_sun_glass_events_WindowEvent_RESIZE,
-                geometry.current_width,
-                geometry.current_height);
-        CHECK_JNI_EXCEPTION(mainEnv)
+    if (moved) {
+        geometry_set_window_x(&geometry, x);
```

Comment:
this will set `windowGeometry->refx` where 2 lines below you set `windowGeometry.current_x`. Unless there is gravity involved, both refer to the same value, one being `int` and the other being `float` though. Is there another conceptual difference, or why do we have 2 values here?

Reply:
I think you're right, I added `current_x/y` to check if a window moved, but `ref_x/ref_y` might contain the same value (different type).

Reply:
Fixed